identity/src/lib.rs

```diff
@@ -115,7 +115,33 @@ impl From<&PublicKey> for proto::PublicKey {
     }
 }
 
+use std::fmt::Display;
+
 pub use error::{DecodingError, SigningError};
 pub use keypair::{Keypair, PublicKey};
 #[cfg(feature = "peerid")]
 pub use peer_id::{ParseError, PeerId};
+
+#[derive(Debug, PartialEq, Eq)]
+/// The type of key a `KeyPair` is holding.
+pub enum KeyType {
+    #[cfg(feature = "ed25519")]
+    Ed25519,
+    #[cfg(all(feature = "rsa", not(target_arch = "wasm32")))]
+    RSA,
+    #[cfg(feature = "secp256k1")]
+    Secp256k1,
+    #[cfg(feature = "ecdsa")]
+    Ecdsa,
+}
```

Comment:
These should not be feature-gated, otherwise we again have a hazard where users might not exhaustively match on them based on the features that are activated. Please remove the `cfg`s here.

Reply:
You've resolved this but I can still see the `cfg`s here. Did you forget to push something?